Distance sensors: scan all busses for sensor and remove nuttx deps

DanielePettenuzzo opened the pull request:
Add -a and -b argument to start distance sensor drivers and search for a device on all I2C busses or to start on a specific bus. Up to now these drivers have a default bus and no possibility to change. This is useful for platforms such as fmu-v5 that has 3 external i2c busses.

Moreover I removed some nuttx dependencies such as the err, errx and exit functions.

I started with the teraranger driver because I have one to test with. Soon I will apply these changes to all distance sensor drivers that use I2C.

hamishwillee commented:
Pretty cool. When this goes in, can you help update https://docs.px4.io/en/sensor/rangefinders.html#teraranger-rangefinders ?

This might also be a good time to add docs in source - e.g. like [this](https://github.com/PX4/Firmware/blob/master/src/drivers/pwm_out_sim/PWMSim.cpp#L646) for generating this [driver doc](http://dev.px4.io/en/middleware/modules_driver.html#pwm_out_sim_usage).

DanielePettenuzzo commented:
@hamishwillee sure! I'll let you know when this is ready to go into master so we can discuss about documentation.

DanielePettenuzzo commented on src/drivers/distance_sensor/mb12xx/mb12xx.cpp:
```diff
@@ -725,64 +771,75 @@ start(uint8_t rotation)
 	}
 
-	errx(1, "driver start failed");
+	PX4_ERR("not started on bus %d", i2c_bus);
```
@bkueng so just have one error if now sensor is found on any bus?

bkueng replied:
Yes, or none at all. Basically just show what hardware is actually found.

DanielePettenuzzo commented on src/drivers/distance_sensor/sf1xx/sf1xx.cpp:
```diff
@@ -637,26 +635,74 @@ namespace sf1xx
-void	test();
-void	reset();
-void	info();
+int bus_options[] = {
```
@bkueng for example in src/lib/drivers/device/i2c.h? (in this case I should rename it since this array exists in some drivers).. or should I create a new i2c_distance.h just common to all distance sensors?

bkueng replied:
> @bkueng for example in src/lib/drivers/device/i2c.h?

yes, and it wil need to have a better name. But it can be used by all I2C drivers.

DanielePettenuzzo commented:
@bkueng I put the bus_option array in src/lib/drivers/device/i2c.h and renamed it to i2c_bus_option. I also changed this for the airspeed sensor drivers.

I also removed all the useless warnings when the sensors were not found on a bus.

bkueng commented on src/lib/drivers/device/i2c.h:
```diff
@@ -37,3 +37,21 @@
 #else
 #include "posix/I2C.hpp"
 #endif
+
```
Can you explicitly include the `board_config.h`?
